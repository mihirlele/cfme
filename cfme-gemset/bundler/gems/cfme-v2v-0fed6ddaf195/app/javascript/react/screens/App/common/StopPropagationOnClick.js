import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'patternfly-react';

// This component has no role on the page, and will be hidden from screen readers (role="presentation").
// Prevents a click on an inner element (e.g. a DropdownKebab) from triggering a click on an ancestor (e.g. a table row).
// The onKeyUp handler and role are used to satisfy eslint-plugin-jsx-a11y.
const StopPropagationOnClick = ({ children }) => (
  <span onClick={e => e.stopPropagation()} onKeyUp={noop} role="presentation">
    {children}
  </span>
);

StopPropagationOnClick.propTypes = {
  children: PropTypes.node
};

export default StopPropagationOnClick;
