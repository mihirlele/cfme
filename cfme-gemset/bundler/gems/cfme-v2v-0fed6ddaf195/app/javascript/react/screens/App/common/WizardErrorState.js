import React from 'react';
import PropTypes from 'prop-types';

const WizardErrorState = ({ title, message, onClose }) => (
  <div className="wizard-pf-complete blank-slate-pf">
    <div className="wizard-pf-success-icon">
      <span className="pficon pficon-error-circle-o" />
    </div>
    <h3 className="blank-slate-pf-main-action">{title}</h3>
    <p className="blank-slate-pf-secondary-action">{message}</p>
    {onClose ? (
      <button type="button" className="btn btn-lg btn-primary" onClick={onClose}>
        {__('Close')}
      </button>
    ) : null}
  </div>
);

WizardErrorState.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string,
  onClose: PropTypes.func
};

WizardErrorState.defaultProps = {
  message: ''
};

export default WizardErrorState;
