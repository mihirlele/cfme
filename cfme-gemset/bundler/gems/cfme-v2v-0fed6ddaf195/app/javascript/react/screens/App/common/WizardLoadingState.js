import React from 'react';
import PropTypes from 'prop-types';
import { Spinner } from 'patternfly-react';

const WizardLoadingState = ({ title, message }) => (
  <div className="wizard-pf-process blank-slate-pf">
    <Spinner loading size="lg" className="blank-slate-pf-icon" />
    <h3 className="blank-slate-pf-main-action">{title}</h3>
    <p className="blank-slate-pf-secondary-action">{message}</p>
  </div>
);

WizardLoadingState.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string
};

WizardLoadingState.defaultProps = {
  message: ''
};

export default WizardLoadingState;
