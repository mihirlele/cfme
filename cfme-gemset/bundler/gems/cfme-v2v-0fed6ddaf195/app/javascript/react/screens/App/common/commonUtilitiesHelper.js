const commonUtilitiesHelper = {
  getMostRecentEntityByCreationDate: entities =>
    entities && entities.reduce((prev, current) => (prev.created_on > current.created_on ? prev : current)),

  groupBy: (items, key) =>
    items &&
    items.reduce(
      (result, item) => ({
        ...result,
        [item[key]]: [...(result[item[key]] || []), item]
      }),
      {}
    ),

  flatten: (arrayItems, flattenedArray) =>
    arrayItems && arrayItems.map(task => task.map(arrayElement => flattenedArray.push(arrayElement)))
};

export default commonUtilitiesHelper;
