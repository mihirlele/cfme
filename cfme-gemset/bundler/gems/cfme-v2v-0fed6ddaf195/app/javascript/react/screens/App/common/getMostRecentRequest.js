const getMostRecentRequest = requests =>
  requests.length && requests.reduce((prev, current) => (prev.created_on > current.created_on ? prev : current));

export default getMostRecentRequest;
