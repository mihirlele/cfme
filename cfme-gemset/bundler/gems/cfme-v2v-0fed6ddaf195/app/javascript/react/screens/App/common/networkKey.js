/**
 * returns a unique network based on current UI model
 */
export const networkKey = network => `${network.clusterId}\\${network.uid_ems || network.ems_ref}`;
