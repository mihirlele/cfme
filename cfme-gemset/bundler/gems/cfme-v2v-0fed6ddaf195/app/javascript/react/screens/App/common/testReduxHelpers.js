import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import configureMockStore from 'redux-mock-store';

const middlewares = [thunk, promiseMiddleware()];

export const generateStore = (reducers, initialState) =>
  createStore(combineReducers(reducers), initialState, applyMiddleware(...middlewares));

export const mockStore = initialState => configureMockStore(middlewares)(initialState);
