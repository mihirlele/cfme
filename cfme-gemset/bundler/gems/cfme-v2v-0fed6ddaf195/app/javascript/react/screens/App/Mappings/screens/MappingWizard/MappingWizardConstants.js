import { OPENSTACK, RHV, V2V_TARGET_PROVIDERS } from '../../../../../../common/constants';

export const V2V_SET_TRANSFORMATIONS_BODY = 'V2V_SET_TRANSFORMATIONS_BODY';
export const V2V_SHOW_WARNING_MODAL = 'V2V_SHOW_WARNING_MODAL';
export const V2V_HIDE_WARNING_MODAL = 'V2V_HIDE_WARNING_MODAL';
export const V2V_SHOW_ALERT = 'V2V_SHOW_ALERT';
export const V2V_HIDE_ALERT = 'V2V_HIDE_ALERT';
export const V2V_MAPPING_WIZARD_EXITED = 'V2V_MAPPING_WIZARD_EXITED';
export const V2V_SHOW_MAPPING_WIZARD = 'V2V_SHOW_MAPPING_WIZARD';
export const V2V_HIDE_MAPPING_WIZARD = 'V2V_HIDE_MAPPING_WIZARD';
export { OPENSTACK, RHV, V2V_TARGET_PROVIDERS };

export const V2V_TARGET_PROVIDER_STORAGE_KEYS = {
  [RHV]: 'storages',
  [OPENSTACK]: 'cloud_volume_types'
};

export const V2V_TARGET_PROVIDER_NETWORK_KEYS = { [RHV]: 'lans', [OPENSTACK]: 'cloud_networks' };

export const TRANSFORMATION_MAPPING_ITEM_SOURCE_TYPES = {
  cluster: 'EmsCluster',
  datastore: 'Storage',
  network: 'Lan'
};

export const TRANSFORMATION_MAPPING_ITEM_DESTINATION_TYPES = {
  [OPENSTACK]: {
    cluster: 'CloudTenant',
    datastore: 'CloudVolumeType',
    network: 'CloudNetwork'
  },
  [RHV]: {
    cluster: 'EmsCluster',
    datastore: 'Storage',
    network: 'Lan'
  }
};
