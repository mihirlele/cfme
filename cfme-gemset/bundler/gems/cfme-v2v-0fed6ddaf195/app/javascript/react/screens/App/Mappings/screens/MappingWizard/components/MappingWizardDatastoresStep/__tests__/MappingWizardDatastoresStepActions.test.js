import { mockStore } from '../../../../../../common/testReduxHelpers';
import { mockRequest, mockReset } from '../../../../../../../../../common/mockRequests';
import { fetchTargetDatastoresAction } from '../MappingWizardDatastoresStepActions';
import {
  targetClusters,
  cloudTenants
} from '../../../../../../../../../redux/common/targetResources/targetResources.fixtures';
import { V2V_TARGET_PROVIDER_STORAGE_KEYS, V2V_TARGET_PROVIDERS } from '../../../MappingWizardConstants';

const store = mockStore({});

afterEach(() => {
  store.clearActions();
  mockReset();
});

describe('fetchTargetDatastoresAction', () => {
  describe('if the target provider is OSP', () => {
    const targetProvider = V2V_TARGET_PROVIDERS[1].id;
    const [targetCloudTenant] = cloudTenants.resources;

    test('dispatches PENDING and FULFILLED actions and adds the provider name', () => {
      mockRequest({
        method: 'GET',
        url: '/api',
        status: 200,
        response: {
          data: {
            ...targetCloudTenant,
            ext_management_system: { name: 'some provider' },
            [V2V_TARGET_PROVIDER_STORAGE_KEYS[targetProvider]]: [{}]
          }
        }
      });

      return store.dispatch(fetchTargetDatastoresAction('/api', '1', targetProvider)).then(() => {
        const actions = store.getActions();
        expect(actions).toMatchSnapshot();
      });
    });
  });

  describe('if the target provider is RHV', () => {
    const targetProvider = V2V_TARGET_PROVIDERS[0].id;
    const [targetCluster] = targetClusters.resources;

    test('dispatches PENDING and FULFILLED actions and adds the provider name', () => {
      mockRequest({
        method: 'GET',
        url: '/api',
        status: 200,
        response: {
          data: {
            ...targetCluster,
            ext_management_system: { name: 'some provider' },
            [V2V_TARGET_PROVIDER_STORAGE_KEYS[targetProvider]]: [{}]
          }
        }
      });

      return store.dispatch(fetchTargetDatastoresAction('/api', '1', targetProvider)).then(() => {
        const actions = store.getActions();
        expect(actions).toMatchSnapshot();
      });
    });
  });

  test('dispatches PENDING and REJECTED actions', () => {
    mockRequest({
      method: 'GET',
      url: '/api',
      status: 404,
      response: { error: 'error' }
    });

    return store.dispatch(fetchTargetDatastoresAction('/api', '1', 'provider')).catch(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
