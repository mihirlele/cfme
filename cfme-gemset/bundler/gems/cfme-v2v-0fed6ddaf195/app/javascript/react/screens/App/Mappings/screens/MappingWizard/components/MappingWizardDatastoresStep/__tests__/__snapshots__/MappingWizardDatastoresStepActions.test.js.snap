// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchTargetDatastoresAction dispatches PENDING and REJECTED actions 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_TARGET_DATASTORES_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: <mocked error>],
    "type": "FETCH_V2V_TARGET_DATASTORES_REJECTED",
  },
]
`;

exports[`fetchTargetDatastoresAction if the target provider is OSP dispatches PENDING and FULFILLED actions and adds the provider name 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_TARGET_DATASTORES_PENDING",
  },
  Object {
    "payload": Object {
      "targetDatastores": Array [
        Object {
          "providerName": "some provider",
        },
      ],
    },
    "type": "FETCH_V2V_TARGET_DATASTORES_FULFILLED",
  },
]
`;

exports[`fetchTargetDatastoresAction if the target provider is RHV dispatches PENDING and FULFILLED actions and adds the provider name 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_TARGET_DATASTORES_PENDING",
  },
  Object {
    "payload": Object {
      "targetDatastores": Array [
        Object {
          "providerName": "some provider",
        },
      ],
    },
    "type": "FETCH_V2V_TARGET_DATASTORES_FULFILLED",
  },
]
`;
